diff --git a/build/rayimg/main.lua b/build/rayimg/main.lua
index 678b39b..9c6ac8f 100644
--- a/build/rayimg/main.lua
+++ b/build/rayimg/main.lua
@@ -62,9 +62,9 @@ local loaded_texture = image_handler:get_current_image()
 local next_texture
 
 
-if args.transition_duration > 0 then
-   next_texture = image_handler:peek_next_image()
-end
+-- if args.transition_duration > 0 then
+--    next_texture = image_handler:peek_next_image()
+-- end
 
 local timer_duration = 0
 local transitioning = false
@@ -106,6 +106,10 @@ while not rl.WindowShouldClose() do
       image_handler:increase_index()
       loaded_texture = image_handler:get_current_image()
       reset_timers()
+      if next_texture ~= nil then
+         rl.UnloadTexture(next_texture.texture)
+         next_texture = nil
+      end
    end
 
    if rl.IsKeyPressed(rl.KEY_LEFT) then
@@ -113,6 +117,15 @@ while not rl.WindowShouldClose() do
       image_handler:decrease_index()
       loaded_texture = image_handler:get_current_image()
       reset_timers()
+      if next_texture ~= nil then
+         rl.UnloadTexture(next_texture.texture)
+         next_texture = nil
+      end
+   end
+
+   if next_texture == nil and args.transition_duration > 0 then
+      drawSingleImage()
+      next_texture = image_handler:peek_next_image()
    end
 
    if args.duration > 0 then
@@ -142,7 +155,8 @@ while not rl.WindowShouldClose() do
          image_handler:increase_index()
          loaded_texture = next_texture
 
-         next_texture = image_handler:peek_next_image()
+         -- next_texture = image_handler:peek_next_image()
+         next_texture = nil
          reset_timers()
       end
    elseif transitioning then
diff --git a/src/rayimg/main.tl b/src/rayimg/main.tl
index 6fb65c3..8464458 100644
--- a/src/rayimg/main.tl
+++ b/src/rayimg/main.tl
@@ -62,9 +62,10 @@ local loaded_texture = image_handler:get_current_image()
 local next_texture: ImageLoader.LoadedTexture
 
 -- will need both images loaded to do a smooth transition
-if args.transition_duration > 0 then
-    next_texture = image_handler:peek_next_image()
-end
+-- will need both images loaded to do a smooth transition
+-- if args.transition_duration > 0 then
+--     next_texture = image_handler:peek_next_image()
+-- end
 
 local timer_duration: number = 0
 local transitioning = false
@@ -106,6 +107,10 @@ while not rl.WindowShouldClose() do
         image_handler:increase_index()
         loaded_texture = image_handler:get_current_image()
         reset_timers()
+        if next_texture ~= nil then
+            rl.UnloadTexture(next_texture.texture)
+            next_texture = nil
+        end
     end
 
     if rl.IsKeyPressed(rl.KEY_LEFT) then
@@ -113,6 +118,15 @@ while not rl.WindowShouldClose() do
         image_handler:decrease_index()
         loaded_texture = image_handler:get_current_image()
         reset_timers()
+        if next_texture ~= nil then
+            rl.UnloadTexture(next_texture.texture)
+            next_texture = nil
+        end
+    end
+
+    if next_texture == nil and args.transition_duration > 0 then
+        drawSingleImage()
+        next_texture = image_handler:peek_next_image()
     end
 
     if args.duration > 0 then
@@ -142,7 +156,8 @@ while not rl.WindowShouldClose() do
             image_handler:increase_index()
             loaded_texture = next_texture
             
-            next_texture = image_handler:peek_next_image()
+            -- next_texture = image_handler:peek_next_image()
+            next_texture = nil
             reset_timers()
         end
     elseif transitioning then
