--- lua-vips/lua-vips-1.1-12.rockspec	2025-09-19 17:31:54.945677178 -0500
+++ lua-vips/lua-vips-1.1-12.rockspec	2025-09-19 17:34:19.434747463 -0500
@@ -23,7 +23,6 @@
 
 dependencies = {
     "lua >= 5.1, < 5.5",   -- standard Lua or LuaJIT >= 2.0
-    "luaffi-tkl >= 1.0"    -- provided by VM with LuaJIT, use `luarocks config rocks_provided.luaffi-tkl 2.1-1` in that case
 }
 
 test_dependencies = {
